qiskit_ibm_runtime/runtime_job.py

```diff
@@ -320,12 +323,42 @@ def _set_error_message(self, job_response: Dict) -> None:
         """
         if self._status == JobStatus.ERROR:
             job_result_raw = self._api_client.job_results(job_id=self.job_id)
-            self._error_message = API_TO_JOB_ERROR_MESSAGE[
```

Comment:
Do we still need the `API_TO_JOB_ERROR_MESSAGE` mapping?

Reply:
I think we'll still need it for places like [here](https://github.com/Qiskit/qiskit-ibm-runtime/blob/main/test/integration/test_job.py#L91) but then there won't be a single source of truth for these values so maybe it's better to still use it in `_error_msg_from_job_response` method 

```
        error_msg = API_TO_JOB_ERROR_MESSAGE["FAILED"]
        if status == "CANCELLED" and self._reason == "Ran too long":
            error_msg = API_TO_JOB_ERROR_MESSAGE["CANCELLED - RAN TOO LONG"]
        return error_msg.format(self.job_id, job_result_raw)
```